ext/web/events.js

```diff
@@ -0,0 +1,164 @@
+export class Event {
+  /** @type {string} */
+  #type;
+  /** @type {boolean} */
+  #bubbles;
+  /** @type {boolean} */
+  #cancelable;
+  /** @type {boolean} */
+  #composed;
+  /** @type {boolean} */
+  #defaultPrevented;
+
+  /**
+   * @param {string} type
+   * @param {EventInit} eventInitDict
+   */
+  constructor(type, eventInitDict = {}) {
+    this.#type = type;
+    this.#bubbles = eventInitDict.bubbles ?? false;
+    this.#cancelable = eventInitDict.cancelable ?? false;
+    this.#composed = eventInitDict.composed ?? false;
+  }
+
+  get type() {
+    return this.#type;
+  }
+
+  // Everything below this is essentially nonsense stubs
+  // Most of these have no relevance in a runtime with no DOM
+  composedPath() {
+    return [];
+  }
+
+  NONE = 0;
+  CAPTURING_PHASE = 1;
+  AT_TARGET = 2;
+  BUBBLING_PHASE = 3;
```

Comment:
AFAIK `NONE`, `CAPTURING_PHASE`, `AT_TARGET` and `BUBBLING_PHASE` should be both available both as static and instance fields
```suggestion
  NONE = 0;
  CAPTURING_PHASE = 1;
  AT_TARGET = 2;
  BUBBLING_PHASE = 3;
  static NONE = 0;
  static CAPTURING_PHASE = 1;
  static AT_TARGET = 2;
  static BUBBLING_PHASE = 3;
```